pymc_marketing/clv/utils.py

```diff
@@ -597,3 +598,185 @@ def rfm_train_test_split(
+        Time granularity for study.
+        Default: 'D' for days. Possible values listed here:
+        https://numpy.org/devdocs/reference/arrays.datetime.html#datetime-units
+    time_scaler: int, optional
+        Default: 1. Useful for scaling recency & T to a different time granularity. Example:
+        With freq='D' and freq_multiplier=1, we get recency=591 and T=632
+        With freq='h' and freq_multiplier=24, we get recency=590.125 and T=631.375
+        This is useful if predictions in a different time granularity are desired,
+        and can also help with model convergence for study periods of many years.
+    sort_transactions: bool, optional
+        Default: True
+        If raw data is already sorted in chronological order, set to `False` to improve computational efficiency.
+
+    Returns
+    -------
+    :obj: DataFrame:
+        customer_id, frequency, recency, T [, monetary_value]
+    """
+
+    rfm_data = rfm_summary(
+        transactions,
+        customer_id_col=customer_id_col,
+        datetime_col=datetime_col,
+        monetary_value_col=monetary_value_col,
+        observation_period_end=observation_period_end,
+        datetime_format=datetime_format,
+        time_unit=time_unit,
+        time_scaler=time_scaler,
+        include_first_transaction=True,
+        sort_transactions=sort_transactions,
+    )
+
+    # change recency to segmentation definition
+    rfm_data["recency"] = rfm_data["T"] - rfm_data["recency"]
+
+    for column_name in zip(
+        ["r_quartile", "f_quartile", "m_quartile"],
+        ["recency", "frequency", "monetary_value"],
+        strict=False,
+    ):
+
+        def _quartile_labels(column_name, max_label_range):
+            # recency labels must be reversed because lower values are more desirable
+            if column_name == "r_quartile":
+                return list(reversed(list(range(1, max_label_range))))
+            else:
+                return range(1, max_label_range)
+
+        try:
+            labels = _quartile_labels(column_name[0], 4)
+            rfm_data[column_name[0]] = pd.qcut(
+                rfm_data[column_name[1]], q=4, labels=labels, duplicates="drop"
+            ).astype(str)
+        except ValueError:
+            try:
+                labels = _quartile_labels(column_name[0], 3)
+                rfm_data[column_name[0]] = pd.qcut(
+                    rfm_data[column_name[1]], q=4, labels=labels, duplicates="drop"
+                ).astype(str)
+            except ValueError:
+                labels = _quartile_labels(column_name[0], 5)
+                rfm_data[column_name[0]] = pd.qcut(
+                    rfm_data[column_name[1]], q=4, labels=labels, duplicates="drop"
+                ).astype(str)
+
+    # TODO: Is `finally` block also needed?
+    rfm_data["rfm_score"] = (
+        rfm_data["r_quartile"] + rfm_data["f_quartile"] + rfm_data["m_quartile"]
+    )
+
+    if segment_config is None:
+        segment_config = {
```

Comment:
Shall we put this default config in a `constants.py` module with an explanation of the meaning in the docstrings . For example

```python
# constants.py

def _get_default_rfm_config() -> dict[str, list[str]]:
    " This default configuration for the RFM model consists of the keys ... "
    return { ... }
```

Reply:
Especially, explaining what does `[334]` mean (for example).

Reply:
Anything else you can think of (including from the MMM side) that can be moved into this `constants.py` module? I support relocating this config into an external variable to clean up the function code, but I'm loathe to create a entirely new module for a single dictionary.